pkg/resourcemanager/controller/health/add.go

```diff
@@ -87,11 +88,16 @@ func AddToManagerWithOptions(mgr manager.Manager, conf ControllerConfig) error {
 				}})
 			},
 		},
-		&conf.ClassFilter, predicate.Or(
+		&conf.ClassFilter,
+		predicate.Or(
 			managerpredicate.ClassChangedPredicate(),
 			// start health checks immediately after MR has been reconciled
 			managerpredicate.ConditionStatusChanged(resourcesv1alpha1.ResourcesApplied, managerpredicate.DefaultConditionChange),
 		),
+		predicate.Or(
+			managerpredicate.NotIgnoreMode(),
+			predicateutils.IsDeleting(),
+		),
```

Comment:
Does this scenario deserve a envtest test case (make sure that the health controller does not perform healthchecks for ManagedResource that is excluded from reconciliation)?

Reply:
good idea, done